src/download.rs

```diff
@@ -92,7 +92,9 @@ pub fn build_agent() -> Result<ureq::Agent> {
     let agent_builder = ureq::builder().user_agent("fuelup");
 
     if let Ok(proxy) = env::var("http_proxy") {
-        return Ok(agent_builder.proxy(ureq::Proxy::new(proxy)?).build());
+        if !proxy.is_empty() {
+            return Ok(agent_builder.proxy(ureq::Proxy::new(proxy)?).build());
+        }
```

Comment:
I think it would be better to surface the error to the user, in case they have any invalid value for `http_proxy`

```suggestion
        return match ureq::Proxy::new(proxy) {
            Some(proxy) => Ok(agent_builder.proxy(proxy).build()),
            Err(err) => {
                error!("Failed to build proxy with http_proxy={proxy}");
                debug!("{:?}", err);
                Err(err)
            }
        };
```

Reply:
> I think it would be better to surface the error to the user, in case they have any invalid value for `http_proxy`

Hi @sdankel the suggestion code will be fail to compile, I have fixed that, thanks for your review *_*